packages/create-angular/src/app/components/views/RiversView.component.ts

```diff
@@ -0,0 +1,363 @@
+
+  histogramTicks = computed(() => {
+    const ticks = [];
+    for (let i = 0; i <= MAX_STREAM_ORDER; i++) {
+      ticks.push(i);
+    }
+    return ticks;
+  });
+
+  /****************************************************************************
+   * Layers (https://deck.gl/docs/api-reference/carto/overview#carto-layers)
+   */
+
+  // Layer visibility represented as name/visibility pairs, managed by the Layers component.
+  layerVisibility = signal<Record<string, boolean>>({
+    [LAYER_ID]: true,
+  });
+
+  // Update layers when data or visualization parameters change.
+  layers = computed(() => [
+    new VectorTileLayer({
+      id: LAYER_ID,
+      pickable: true,
+      visible: this.layerVisibility()[LAYER_ID],
+      data: this.data(),
+      getLineColor: (d) => {
+        return streamOrderToColor(d.properties.streamOrder, colors) as Color;
+      },
+      getLineWidth: (d) => {
+        return Math.pow(d.properties.streamOrder, 2);
+      },
+      lineWidthScale: 20,
+      lineWidthUnits: 'meters',
+      lineWidthMinPixels: 1,
+      onViewportLoad: (tiles) => {
+        const data = this.data();
+        if (data) {
+          data.then((res) => {
+            this.tilesLoaded.set(true);
+            res.widgetSource.loadTiles(tiles);
+            this.viewState.set({ ...this.viewState() });
+          });
+        }
+      },
+    }),
+  ]);
+
+  /****************************************************************************
+   * Legends
+   */
+
+  legendValues = computed(() => {
+    return Array.from({ length: 10 }, (_, i) => (i + 1).toString());
+  });
+
+  getSwatchColor = (value: string) => {
+    return streamOrderToColor(Number(value), colors) as Color;
+  };
+
+  /****************************************************************************
+   * Effects (https://angular.dev/guide/signals#effects)
+   */
+
+  // Update the map view when the viewstate ref changes.
+  private viewStateEffect = effect(() => {
+    const { longitude, latitude, ...rest } = this.viewState();
+    this.map.jumpTo({ center: [longitude, latitude], ...rest });
+  });
+
+  // Update the deck layers when the layers ref change.
+  private layersEffect = effect(() => {
+    this.deck.setProps({ layers: this.layers() });
+  });
+
+  // Update the attribution HTML and zoom data when the data ref changes.
+  private attributionEffect = effect(() => {
+    this.data().then(({ attribution }: { attribution: string }) =>
+      this.attributionHTML.set(attribution),
+    );
+  });
+
+  private extractEffect = effect(() => {
+    const data = this.data();
+    if (this.tilesLoaded() && this.viewStateDebounced()) {
+      data.then((res) => {
+        const bbox = new WebMercatorViewport(
+          this.viewStateDebounced(),
+        ).getBounds();
+        const spatialFilter = createViewportSpatialFilter(bbox);
+        if (spatialFilter) {
+          res.widgetSource.extractTileFeatures({ spatialFilter });
+        }
+      });
+    }
+  });
```

Comment:
No action needed now, but I think this effect can be removed after https://github.com/CartoDB/carto-api-client/pull/103 is merged and published.

Reply:
Yes, this can be improved after filters have changed. If you can release an alpha with the changes of that PR, I can start testing it here.